File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int index1=0;
        int index2 = numbers.size()-1;
        int sum = numbers[index1] + numbers[index2];
        while(index1<index2 && sum!=target){
            sum = numbers[index1] + numbers[index2];
            if(sum>target){
                index2--;
            }
            else if(sum<target){
                index1++;
            }
        }
        return {index1+1, index2+1};
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-1.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        sub = []
        def dfs(i):
            a = sum(sub)
            if a>=target or i>=len(nums):
                if a==target:
                    res.append(sub.copy())
                return
            sub.append(nums[i])
            dfs(i)
            sub.pop()
            dfs(i+1)
        dfs(0)
        return res
            